t/dataset.t

```diff
@@ -0,0 +1,196 @@
+use v6;
+use Agrammon::Config;
+use Agrammon::DB::Dataset;
+use Agrammon::DB::User;
+use Agrammon::Web::SessionUser;
+use DB::Pg;
+use Test;
+
+plan 11;
+
+if %*ENV<AGRAMMON_UNIT_TEST> {
+    skip-rest 'Not a unit test';
+    exit;
+}
+
+my $*AGRAMMON-DB-CONNECTION;
+
+subtest 'Connect to database' => {
+    my $conninfo;
+    my $cfg-file;
+    if %*ENV<DRONE_REPO> {
+        $cfg-file = %*ENV<AGRAMMON_CFG> // "t/test-data/agrammon.drone.cfg.yaml";
+    }
+    else {
+        $cfg-file = %*ENV<AGRAMMON_CFG> // "t/test-data/agrammon.cfg.yaml";
+    }
+    my $cfg = Agrammon::Config.new;
+    ok $cfg.load($cfg-file), "Load config from file $cfg-file";
+    $conninfo = $cfg.db-conninfo;
+    ok $*AGRAMMON-DB-CONNECTION = DB::Pg.new(:$conninfo), 'Create DB::Pg object';
+}
+
+my $user;
+subtest 'Create user' => {
+    ok $user = Agrammon::DB::User.new(
+        :username<agtestuser>,
+    ), 'Create new user';
+    is $user.username, 'agtestuser', 'User has correct username';
+}
+
+subtest 'Create dataset' => {
+    my $date = DateTime.new('2020-04-01T00:00:00Z');
+    diag $date;
+    ok my $dataset = Agrammon::DB::Dataset.new(
+        :id<42>,
+        :name<agtest>,
+        :read-only,
+        :model<TestModel>,
+        :comment('Test comment'),
+        :version('6.0-test'),
+        :records(42),
+        :mod-date($date),
+        :data(),
+        :tags()
+        :user($user)
+    ), 'Create new dataset';
+    is $dataset.name,     'agtest', 'User has correct username';
+    is $dataset.read-only, True,     'Dataset is read-only';
+    is $dataset.model,     'TestModel',     'User has correct model';
+    is $dataset.comment,   'Test comment',     'User has correct comment';
+    is $dataset.version,     '6.0-test',     'User has correct version';
+    is $dataset.records,     42,     'User has correct records';
+    is $dataset.mod-date,    $date,     'User has correct mod-date';
+    is $dataset.data.elems,  0,     'User has 0 data';
+    is $dataset.tags.elems,  0,     'User has 0 tags';
+#    is $dataset.user.username,   'agtestuser',     'User has correct user';
+}
+
+# done-testing; exit;
```

Comment:
Leftover things to clean up?

Reply:
And a bad one ... prevented the tests from running ... at least I had a plan ... I just wonder way GHA didn't fail.